src/reader.rs

```diff
@@ -4204,6 +4204,11 @@ impl<'a> Reader<'a> {
         // hack to work.
         reader_write_title(L!(""), zelf.parser, false);
 
+        // Reap jobs but do NOT trigger a repaint.
+        // This is to prevent infinite loops in case a job from the prompt triggers a repaint.
+        // See #9796.
+        job_reap(zelf.parser, true);
+
```

Comment:
LGTM, this will fix it for foreground jobs in the prompt, which is the most common scenario (I can't think of a legit case for a background job that's not disowned).